folium/folium.py

```diff
@@ -251,7 +252,10 @@ def choropleth(self, geo_path=None, geo_str=None, data_out='data.json',
 
         # Create GeoJson object
         if geo_path:
-            geo_data = open(geo_path)
+            if geo_path.lower().startswith(('http', 'ftp', 'https')):
```

Comment:
Maybe it is an overkill but how about something like:

```python
def scheme_validator(url):
    parsed = urlparse(url)
    if parsed.scheme in ['http', 'ftp', 'https']:
        return True
    return False
```

to validade the scheme?